src/idl_gen_json_schema.cpp

```diff
@@ -20,205 +20,272 @@
-      typeline.append("\"type\" : \"array\", \"items\" : { ");
-      if (type.element == BASE_TYPE_STRUCT) {
-        typeline.append(GenTypeRef(type.struct_def));
-      } else {
-        typeline.append(GenType(GenNativeType(type.element)));
-      }
-      typeline.append(" }");
-      return typeline;
-    }
-    case BASE_TYPE_STRUCT: {
-      return GenTypeRef(type.struct_def);
-    }
-    case BASE_TYPE_UNION: {
-      std::string union_type_string("\"anyOf\": [");
-      const auto &union_types = type.enum_def->vals.vec;
-      for (auto ut = union_types.cbegin(); ut < union_types.cend(); ++ut) {
-        auto &union_type = *ut;
-        if (union_type->union_type.base_type == BASE_TYPE_NONE) {
-          continue;
-        }
-        if (union_type->union_type.base_type == BASE_TYPE_STRUCT) {
-          union_type_string.append("{ " + GenTypeRef(union_type->union_type.struct_def) + " }");
-        }
-        if (union_type != *type.enum_def->vals.vec.rbegin()) {
-          union_type_string.append(",");
-        }
+  namespace jsons {
+
+    std::string GenNativeType(BaseType type) {
+      switch (type) {
+      case BASE_TYPE_BOOL:
+          return "boolean";
+      case BASE_TYPE_CHAR:
+      case BASE_TYPE_UCHAR:
+      case BASE_TYPE_SHORT:
+      case BASE_TYPE_USHORT:
+      case BASE_TYPE_INT:
+      case BASE_TYPE_UINT:
+      case BASE_TYPE_LONG:
+      case BASE_TYPE_ULONG:
+      case BASE_TYPE_FLOAT:
+      case BASE_TYPE_DOUBLE:
+        return "number";
+      case BASE_TYPE_STRING:
+        return "string";
+      default:
+        return "";
       }
-      union_type_string.append("]");
-      return union_type_string;
     }
-    case BASE_TYPE_UTYPE:
-      return GenTypeRef(type.enum_def);
-    default:
-      return GenType(GenNativeType(type.base_type));
-  }
-}
-
-class JsonSchemaGenerator : public BaseGenerator {
- private:
-  CodeWriter code_;
-
- public:
-  JsonSchemaGenerator(const Parser &parser, const std::string &path,
-                      const std::string &file_name)
-      : BaseGenerator(parser, path, file_name, "", "") {}
-
-  explicit JsonSchemaGenerator(const BaseGenerator &base_generator)
-      : BaseGenerator(base_generator) {}
-
-  bool generate() {
-    code_.Clear();
-    code_ += "{";
-    code_ += "  \"$schema\": \"http://json-schema.org/draft-04/schema#\",";
-    code_ += "  \"definitions\": {";
-    for (auto e = parser_.enums_.vec.cbegin();
-         e != parser_.enums_.vec.cend();
-         ++e) {
-      code_ += "    \"" + GenFullName(*e) + "\" : {";
-      code_ += "      " + GenType("string") + ",";
-      std::string enumdef("      \"enum\": [");
-      for (auto enum_value = (*e)->vals.vec.begin(); 
-           enum_value != (*e)->vals.vec.end();
-           ++enum_value) {
-        enumdef.append("\"" + (*enum_value)->name + "\"");
-        if (*enum_value != (*e)->vals.vec.back()) {
-          enumdef.append(", ");
+
+  class JsonSchemaGenerator : public BaseGenerator {
```

Comment:
I agree.  the original diff was basically a -file +file.  I had to go line by line to to get it resolved to this point.  (original code source was from pre-google-formatting, used more legible for loops, strstream, etc)
Feel free to apply your own linter/formatter.